### PR TITLE
[SPARK-38121][PYTHON][SQL][FOLLOW-UP] Set 'spark.sql.catalogImplementation' to 'hive' in HiveContext

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR is a followup of https://github.com/apache/spark/pull/35410 to fix a mistake. `HiveContext` should set `spark.sql.catalogImplementation` to `hive` instead of `in-memory`.

This PR also includes several changes:
- Make `HiveContext.getOrCreate` works identically as `SQLContext.getOrCreate`
- Match the signature of `HiveContext.__init__` and `SQLContext.__init__` (both are not supported to be directly called by users though).

### Why are the changes needed?

See https://github.com/apache/spark/pull/35410#discussion_r806358814

### Does this PR introduce _any_ user-facing change?

No to end users because this change has not been released out yet.

It creates a non-Hive supported `SparkSession` if there isn't an existing SparkSession running. See also https://github.com/apache/spark/pull/35410#discussion_r806358814.

Nobody uses `HiveContext` directly anymore but it's better to keep the behaviours unchanged.

### How was this patch tested?

Manually tested (in PySpark shell):

```python
spark.stop()
from pyspark import SparkContext
from pyspark.sql import HiveContext
HiveContext.getOrCreate(SparkContext.getOrCreate()).getConf("spark.sql.catalogImplementation")
```

**Before:**

```pyspark
'in-memory'
```

**After:**

```pyspark
'hive'
```

### CODE REVIEW

File: python/pyspark/sql/context.py [hunk 1]
```diff
@@ -705,19 +707,33 @@ class HiveContext(SQLContext):
 
     """
 
-    def __init__(self, sparkContext: SparkContext, jhiveContext: Optional[JavaObject] = None):
+    _static_conf = {"spark.sql.catalogImplementation": "hive"}
+
+    def __init__(
```
Comment (HyukjinKwon) [inline python/pyspark/sql/context.py:712]: This matches the signature w/ `SQLContext.__init__`

File: python/pyspark/sql/context.py [hunk 2]
```diff
@@ -705,19 +707,33 @@ class HiveContext(SQLContext):
...
-        SQLContext.__init__(self, sparkContext, session, jhiveContext)
+        if sparkSession is not None:
+            sparkSession = SparkSession._getActiveSessionOrCreate(**static_conf)
+        SQLContext.__init__(self, sparkContext, sparkSession, jhiveContext)
+
+    @classmethod
+    def _get_or_create(
+        cls: Type["SQLContext"], sc: SparkContext, **static_conf: Any
+    ) -> "SQLContext":
+        return SQLContext._get_or_create(sc, **HiveContext._static_conf)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/context.py:736]: This is to make sure `HiveContext.getOreCreate` sets `spark.sql.catalogImplementation` correctly.

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Thanks. Merging to master.

**Comment by HyukjinKwon** [comment]
Thanks!!!!

**Pull request closed without merging**